model/task/task.go

```diff
@@ -4286,3 +4289,66 @@ func (t *Task) GetEstimatedCost(ctx context.Context) (TaskCost, error) {
 	runtimeSeconds := estimatedDuration.Seconds()
 	return CalculateTaskCost(runtimeSeconds, costData, financeConfig), nil
 }
+
+func (task *Task) MoveLogsByNamesToBucket(ctx context.Context, settings *evergreen.Settings, output *TaskOutput) error {
+	if output.TestLogs.BucketConfig != output.TaskLogs.BucketConfig {
+		// test logs and task logs will always be in the same bucket
+		return errors.New("test log and task log buckets do not match")
+	}
+	srcBucket, err := newBucket(ctx, output.TestLogs.BucketConfig, output.TestLogs.AWSCredentials)
+	if err != nil {
+		return errors.Wrap(err, "getting regular test log bucket")
+	}
+
+	logService := log.NewLogServiceV0(srcBucket)
+
+	logNames := make([]string, 0, 4)
+	// add task logs
+	for _, logType := range []TaskLogType{TaskLogTypeAgent, TaskLogTypeSystem, TaskLogTypeTask} {
+		logNames = append(logNames, getLogName(*task, logType, output.TaskLogs.ID()))
+	}
+	// add test logs
+	logNames = append(logNames, fmt.Sprintf("%s/%s/%d/%s", task.Project, task.Id, task.Execution, output.TestLogs.ID()))
+
+	keys, err := logService.GetChunkKeys(ctx, logNames)
+	if err != nil {
+		return errors.Wrap(err, "getting chunk keys for log names")
+	}
+	allKeys := keys
+
+	failedCfg := settings.Buckets.LogBucketFailedTasks
+	failedBucket, err := newBucket(ctx, failedCfg, output.TestLogs.AWSCredentials)
+	if err != nil {
+		return errors.Wrap(err, "getting failed bucket")
+	}
+
+	if err = logService.MoveObjectsToBucket(ctx, allKeys, failedBucket); err != nil {
+		return errors.Wrap(err, "moving logs to failed bucket")
+	}
+
+	task.TaskOutputInfo.TaskLogs.BucketConfig = failedCfg
+	task.TaskOutputInfo.TestLogs.BucketConfig = failedCfg
```

Comment:
How come we do this at the end? It looks like a random sideeffect (maybe we should split it to another func or just make it in the name or the comment might be enough)